src/main/java/com/bandwidth/sqs/consumer/strategy/expiration/VisibilityTimeoutPercentageExpiration.java

```diff
@@ -0,0 +1,23 @@
+package com.bandwidth.sqs.consumer.strategy.expiration;
+
+import com.bandwidth.sqs.queue.SqsMessage;
+
+import java.time.Duration;
+
+/**
+ * Considers a message expired after it's age is a certain percentage of the visibility timeout
+ */
+public class VisibilityTimeoutPercentageExpiration implements ExpirationStrategy {
+
+    private final double percent;
+
+    public VisibilityTimeoutPercentageExpiration(double percent) {
+        this.percent = percent;
+    }
+
+    @Override
+    public boolean isExpired(SqsMessage<?> sqsMessage, Duration visibilityTimeout) {
```

Comment:
Couldn't the visibility timeout just be in the constructor of this implementation? Do we expect it to change during runtime?

Reply:
This strategy could be used for multiple queues (with different visibility timeouts)

Reply:
And in the future it would be good to detect changes in visibility timeout, and these could be applied to the expiration strategy automatically

Reply:
I would assume that if you were using it with multiple queues, you would just make a separate instance of the `ExpirationStrategy` for each one. If you could detect changes in it then it makes more sense, but it might seem better to have something able to set it on the class instead of passing it to each call, if that matters enough